cmds/time_sos/service.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"os/exec"
+	"time"
+)
+
+type TimeService struct {
+	Date string
+	Time string
+}
+
+func getCurrentDate() string {
+	return time.Now().Format("2006-01-02")
+}
+
+func getCurrentTime() string {
+	return time.Now().Format("15:04")
+}
+
+func (ts *TimeService) Update() {
+	// Update sets the time service to the current system time
+	ts.Date = getCurrentDate()
+	ts.Time = getCurrentTime()
+}
+
+func (ts TimeService) AutoSetTime() error {
+	// AutoSetTime calls ntpdate to get the current date from time.google.com.
+	// TODO: Find a go ntp package to remove dependence on outside function
+	return exec.Command("ntpdate").Run()
```

Comment:
Go ahead and remove the todo. 

We need to decide whether to import the ntpdate code here, or what to do. Not sure.